### ISSUE TITLE
bug: own-methods-must-be-private enforces `private` accessor on `#name` while typescript disallowes it

### ISSUE THREAD

**Issue by mt3o** [stencil-community/stencil-eslint#125]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/stencil-community/stencil-eslint/issues) that already report this problem, without success.

### Stencil ESLint Version

0.8.0

### Current Behavior

When using fields and methods marked with private identifier # like `#methodName` we get warning from eslint, triggering the rule
`ESLint: Own class methods cannot be public(@stencil-community/ own-methods-must-be-private)`, the fix is to add the `private` modifier. This on the other hand fails with typescript `TS18010: An accessibility modifier cannot be used with a private identifier.`

### Expected Behavior

Have eslint recognize names starting with # as being private, and don't report violation. 

### Steps to Reproduce

Have a simple component like:
```
@Component({/*...*/})
class foo{
#someName(){ /*...*/ }
}
```
Eslint will try to add the `private` keyword, but then typescript reports invalid syntax.

### Code Reproduction URL

no url

**Comment by james-nash** [comment]
Just encountered this exact issue too.
